istio/install.go

```diff
@@ -37,18 +57,124 @@ func (istio *Istio) installIstio(del bool, version, namespace string) (string, e
-		return st, ErrInstallIstio(err)
+		return st, ErrApplyHelmChart(err)
 	}
 
+	// TODO: keep this as a fallback method
+	//err = istio.runIstioCtlCmd(version, del)
+	//if err != nil {
+	//	istio.Log.Error(ErrInstallIstio(err))
+	//	return st, ErrInstallIstio(err)
+	//}
+
 	if del {
 		return status.Removed, nil
 	}
 	return status.Installed, nil
 }
 
+func (istio *Istio) applyHelmChart(del bool, version, namespace string) error {
+	kClient := istio.MesheryKubeclient
+
+	// charts should be here ideally
+	repo := "https://gcsweb.istio.io/gcs/istio-release/releases/charts/"
+
+	chart := "istio"
+
+	chartVersion, err := ConvertAppVersionToChartVersion(repo, chart, version)
+	if err != nil {
+		return ErrConvertingAppVersionToChartVersion(err)
+	}
+
+	err = kClient.ApplyHelmChart(mesherykube.ApplyHelmChartConfig{
+		ChartLocation: mesherykube.HelmChartLocation{
+			Repository: repo,
+			Chart:      chart,
+			Version:    chartVersion,
+		},
+		Namespace:       namespace,
+		Delete:          del,
+		CreateNamespace: true,
+	})
+
+	return err
+}
+
+// ConvertAppVersionToChartVersion takes in the repo, chart and app version and
+// returns the corresponding chart version for the same
+func ConvertAppVersionToChartVersion(repo, chart, appVersion string) (string, error) {
+	appVersion = normalizeVersion(appVersion)
+
+	helmIndex, err := CreateHelmIndex(repo)
+	if err != nil {
+		return "", ErrCreatingHelmIndex(err)
+	}
+
+	entryMetadata, exists := helmIndex.Entries.GetEntryWithAppVersion(chart, appVersion)
+	if !exists {
+		return "", ErrEntryWithAppVersionNotExists(chart, appVersion)
+	}
+
+	return entryMetadata.Version, nil
+}
+
+// CreateHelmIndex takes in the repo name and creates a
+// helm index for it. Helm index is basically marshalled version of
+// index.yaml file present in the remote helm repository
+func CreateHelmIndex(repo string) (*HelmIndex, error) {
+	url := fmt.Sprintf("%s/index.yaml", repo)
+
+	// helm repository path will alaways be varaible hence,
+	// #nosec
+	resp, err := http.Get(url)
+	if err != nil || resp.StatusCode != http.StatusOK {
+		return nil, ErrHelmRepositoryNotFound(repo, err)
+	}
+	defer func() {
+		_ = resp.Body.Close()
+	}()
+
+	var hi HelmIndex
+	dec := yaml.NewDecoder(resp.Body)
+	if err := dec.Decode(&hi); err != nil {
+		return nil, ErrDecodeYaml(err)
+	}
+
+	return &hi, nil
+}
+
+// GetEntryWithAppVersion takes in the entry name and the appversion and returns the corresponding
+// metadata for the parameters if it exists
+func (helmEntries HelmEntries) GetEntryWithAppVersion(entry, appVersion string) (HelmEntryMetadata, bool) {
+	hem, ok := helmEntries[entry]
+	if !ok {
+		return HelmEntryMetadata{}, false
+	}
+
+	for _, v := range hem {
+		if v.Name == entry && v.AppVersion == appVersion {
+			return v, true
+		}
+	}
+
+	return HelmEntryMetadata{}, false
+}
+
+// normalizeVerion takes in a version and adds "v" prefix
+// if it isn't already present
+func normalizeVersion(version string) string {
```

Comment:
The version string used is that of the release's git tag, correct?

Reply:
Yes, the git tag of the latest release. There can be two versions fields in helm charts
I think `AppVersion` is generally the git tag https://github.com/openservicemesh/osm/blob/7b40684d291691dac7374c70ae647889e6494b48/charts/osm/Chart.yaml#L21 
and `Version` is the version of the chart (doesn't contain a v prefix)